### PR TITLE
Feature/shadow override improvements

### PR DESCRIPTION
**Pull request by simonmitchell**
Adds `UView+ShadowComponents` extension that `ThunderCloud` also uses to make managing shadows much easier!

### CODE REVIEW

File: MessageStackView/Views/Badges/BadgeContainerView.swift [hunk 1]
```diff
@@ -14,6 +14,9 @@ import UIKit
 /// Inherit `ShakeView` to add badge animations
 open class BadgeContainerView: UIView {
     
+    /// Overrideable default shadow components for `BadgeContainerView`
+    static var shadowComponents: UIView.ShadowComponents = .default
```
Comment (BenCShutt) [inline MessageStackView/Views/Badges/BadgeContainerView.swift:18]: Let's just this in `Constants`.
In a lot of projects I make constants like
```
// MARK: - Constants

extension UIView.ShadowComponents {
    static let shadowComponents: UIView.ShadowComponents = .default
}
```

But for this project I follows the theme of a `Constants` `struct` inside the `class`/`struct`

File: MessageStackView/Extensions/UIView+ShadowComponents.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            offset: .zero
+        )
+    }
+    
+    var shadow: ShadowComponents? {
+        set {
+            guard let newValue = newValue else {
+                layer.shadowRadius = 0
+                layer.shadowOpacity = 0
+                layer.shadowColor = nil
```
Comment (BenCShutt) [inline MessageStackView/Extensions/UIView+ShadowComponents.swift:59]: The defaults of these properties in UIKit are a bit different I think. Up to you whether you think we should use the UIKit defaults or these values @simonmitchell

File: MessageStackView/Views/Badges/BadgeContainerView.swift (hunk 1 above)
Comment (BenCShutt) [reply MessageStackView/Views/Badges/BadgeContainerView.swift:18]: Maybe you have gone this way for overridable reasons. In that case IMO the default should be a constant (as you've done). But the instance should be able to override? Happy either way here!

File: MessageStackView/Extensions/UIView+ShadowComponents.swift (hunk 2 above)
Comment (simonmitchell) [reply MessageStackView/Extensions/UIView+ShadowComponents.swift:59]: Will update quickly! Are the defaults the ones used in your extension? :)

Comment (BenCShutt) [reply MessageStackView/Extensions/UIView+ShadowComponents.swift]: @simonmitchell Prob best to quickly looking up the docs with `alt` click might be safest

**Pull request merged**